src/clusterfuzz/_internal/cron/external_testcase_reader.py

```diff
@@ -0,0 +1,149 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Automated ingestion of testcases via IssueTracker."""
+
+import re
+
+import requests
+
+from appengine.libs import form
+from appengine.libs import gcs
+from appengine.libs import helpers
+from clusterfuzz._internal.issue_management.google_issue_tracker import \
+    issue_tracker
+
+ACCEPTED_FILETYPES = [
+    'text/javascript', 'application/pdf', 'text/html', 'application/zip'
+]
+
+
+def close_invalid_issue(upload_request, attachment_info, description):
+  """Closes any invalid upload requests with a helpful message."""
+  comment_messsage = (
+      'Hello, this issue is automatically closed. Please file a new bug after'
+      'fixing the following issues:\n\n')
+  invalid = False
+
+  # TODO(pgrace) remove after testing.
+  if upload_request.id == '373893311':
+    return False
+
+  # TODO(pgrace) add secondary check for authorized reporters.
+
+  # Issue must have exactly one attachment.
+  if len(attachment_info) != 1:
+    comment_messsage += 'Please provide exactly one attachment.\n'
+    invalid = True
+  else:
+    # Issue must use one of the supported testcase file types.
+    if attachment_info[0]['contentType'] not in ACCEPTED_FILETYPES:
+      comment_messsage += (
+          'Please provide an attachment of type: html, js, pdf, or zip.\n')
+      invalid = True
+    if not attachment_info[0]['attachmentDataRef'] or \
+      not attachment_info[0]['attachmentDataRef']['resourceName'] \
+        or not attachment_info[0]['filename']:
+      comment_messsage += \
+        'Please check that the attachment uploaded successfully.\n'
+      invalid = True
+
+  # Issue must have valid flags as the description.
+  flag_format = re.compile(r'^([ ]?\-\-[A-Za-z\-\_]*){50}$')
+  if flag_format.match(description):
+    comment_messsage += (
+        'Please provide flags in the format: "--test_flag_one --testflagtwo",\n'
+    )
+    invalid = True
+
+  if invalid:
+    comment_messsage += (
+        '\nPlease see the new bug template for more information on how to use'
+        'Clusterfuzz direct uploads.')
+    upload_request.status = 'not_reproducible'
+    upload_request.save(new_comment=comment_messsage, notify=True)
+
+  return invalid
+
+
+def submit_testcase(issue_id, file, filename, filetype, cmds):
+  """Uploads the given testcase file to Clusterfuzz."""
+  if filetype == 'text/javascript':
+    job = 'linux_asan_d8_dbg'
+  elif filetype == 'application/pdf':
+    job = 'libfuzzer_pdfium_asan'
+  elif filetype == 'text/html':
+    job = 'linux_asan_chrome_mp'
+  elif filetype == 'application/zip':
+    job = 'linux_asan_chrome_mp'
+  else:
+    raise TypeError
+  upload_info = gcs.prepare_blob_upload()._asdict()
```

Comment:
I think we should find a way of using this without accessing private methods.

Reply:
is _asdict() private? i thought it was one of those weird python things that were public